dask/backends.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from functools import lru_cache
+from typing import Callable
+
+from dask import config
+from dask.compatibility import entry_points
+
+
+class DaskBackendEntrypoint(ABC):
+    """Base Collection-Backend Entrypoint Class
+
+    Most methods in this class correspond to collection-creation
+    for a specific library backend. Once a collection is created,
+    the existing data will be used to dispatch compute operations
+    within individual tasks. The backend is responsible for
+    ensuring that these data-directed dispatch functions are
+    registered when ``__init__`` is called.
+    """
+
+    @abstractmethod
+    def __init__(self):
+        """Register data-directed dispatch functions
+
+        This may be a ``pass`` operation if the data dispatch functions
+        are already registered within the same module that the
+        ``DaskBackendEntrypoint`` subclass is defined.
+        """
+        raise NotImplementedError
+
+
+@lru_cache(maxsize=1)
+def detect_entrypoints():
+    entrypoints = entry_points("dask.backends")
+    return {ep.name: ep for ep in entrypoints}
+
+
+class CreationDispatch:
+    """Simple backend dispatch for collection-creation functions"""
+
```

Comment:
One could just populate the entrypoints at class definition time:
```python
    entrypoints = {ep.name: ep for ep in entry_points("dask.backends")}
```

Reply:
Hmm, I kind of like the idea that we don't take any time to search for external entrypoints if the user is only working with internally-registereed backends. (not sure at all if this is a reasonable concern)

Reply:
Sure.